src/reactpy/core/vdom.py

```diff
@@ -148,7 +148,7 @@ def __call__(
     ) -> VdomDict:
         """The entry point for the VDOM API, for example reactpy.html(<WE_ARE_HERE>)."""
         attributes, children = separate_attributes_and_children(attributes_and_children)
-        key = attributes.pop("key", None)
+        key = attributes.get("key", None)
```

Comment:
Per your suggestion, how does this look? Test passed. Seems like it should maintain all of the same behavior and is a cleaner solution.